lib/cpp/src/thrift/transport/TSSLSocket.cpp

```diff
@@ -471,8 +471,10 @@ void TSSLSocket::checkHandshake() {
       }
     } while (rc == 2);
   } else {
-    // set the SNI hostname
-    SSL_set_tlsext_host_name(ssl_, getHost().c_str());
+    /* OpenSSL < 0.9.8f does not have SSL_set_tlsext_host_name() */
+    #if defined(SSL_set_tlsext_host_name) // set the SNI hostname
```

Comment:
Typically we would add something to the build system environment to differentiate this; also is there an alternative that can be used with older OpenSSL?  Other folks who are committers will need to decide if it is worth supporting an older and likely quite vulnerable (to hacks) OpenSSL library.